src/components/SourceCodeEditor.tsx

```diff
@@ -24,61 +25,65 @@ const SourceCodeEditor = (props: ISourceCodeEditor) => {
   const editorRef = useRef<Parameters<EditorDidMount>[0] | null>(null);
   const { sourceCode, setSourceCode } = useIntegrationSourceStore();
   const [syncedCode, setSyncedCode] = useState('');
-  const { integrationJson, updateIntegration } = useIntegrationJsonStore((state) => state);
   const { settings, setSettings } = useSettingsStore();
-  const previousJson = usePrevious(integrationJson);
+  const { flows, properties, setFlows } = useFlowsStore(({ flows, properties, setFlows }) => ({ flows, properties, setFlows }), shallow);
+  const previousFlows = usePrevious(flows);
+
   const schemaUri = settings.dsl.validationSchema
     ? process.env.KAOTO_API + settings.dsl.validationSchema
     : '';
 
   useEffect(() => {
-    let tmpInt: IIntegration = integrationJson;
```

Comment:
And the content in this `useEffect` would be also a candidate to move out to a `FlowsService`